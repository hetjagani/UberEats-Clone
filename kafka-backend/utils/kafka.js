const sendData = (producer, data, isError) => {
  producer.send(
    [{ topic: 'response', messages: JSON.stringify({ data, isError }), partitions: 1 }],
    (err, d) => {
      if (err) {
        console.error(err);
        throw Error('error sending data to kafka');
      } else {
        console.log('SENT DATA:', d);
      }
    },
  );
};

module.exports = {
  sendData,
};
