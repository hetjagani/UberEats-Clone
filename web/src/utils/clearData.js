export const clearData = () => {
  document.cookie.split(';').forEach(function (c) {
    document.cookie = c
      .replace(/^ +/, '')
      .replace(/=.*/, '=;expires=' + new Date().toUTCString() + ';path=/');

    window.localStorage.removeItem('persist:root');
  });
};
