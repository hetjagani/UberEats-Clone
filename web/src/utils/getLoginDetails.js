import jwt from 'jsonwebtoken';
import { getCookie } from 'react-use-cookie';

const getLoginDetails = () => {
  const token = getCookie('auth');
  if (token) {
    const tokendata = jwt.decode(token);
    return tokendata;
  } else {
    return {};
  }
};

export default getLoginDetails;
