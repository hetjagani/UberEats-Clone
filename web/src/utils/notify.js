import { toast } from 'react-toastify';

const notify = ({ type, description }) => {
  toast[type](`${type}: ${description}`, {
    position: 'top-center',
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,
  });
};

export default notify;
