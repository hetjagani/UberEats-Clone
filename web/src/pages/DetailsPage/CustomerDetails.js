import React, { useEffect, useState } from 'react';
import { FormControl } from 'baseui/form-control';
import { Input, MaskedInput } from 'baseui/input';
import { Textarea } from 'baseui/textarea';
import { useStyletron } from 'baseui';
import { Select, StatefulSelect } from 'baseui/select';
import { Button, SIZE } from 'baseui/button';
import { useDispatch, useSelector } from 'react-redux';
import {
  clearCustomerMedium,
  createCustomer,
  createCustomerMedium,
  updateCustomer,
} from '../../actions/customers';
import { Display4 } from 'baseui/typography';
import { Card } from 'baseui/card';
import { FileUploader } from 'baseui/file-uploader';
import S3 from 'react-aws-s3';
import notify from '../../utils/notify';
import { useHistory } from 'react-router';

function CustomerDetails({ loginDetails, update }) {
  const [css, theme] = useStyletron();
  const formContainer = css({
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    width: '80%',
  });
  const mainContainer = css({
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
  });
  const imgContainer = css({
    marginLeft: '50px',
  });

  const cityOpts = [
    { id: 'San Jose', city: 'San Jose' },
    { id: 'Berkely', city: 'Berkely' },
    { id: 'Los Angeles', city: 'Los Angeles' },
    { id: 'San Diego', city: 'San Diego' },
    { id: 'San Francisco', city: 'San Francisco' },
    { id: 'Fresno', city: 'Fresno' },
    { id: 'Sacremento', city: 'Sacremento' },
  ];

  const stateOpts = [
    { id: 'California', state: 'California' },
    { id: 'Arizona', state: 'Arizona' },
    { id: 'Colorado', state: 'Colorado' },
  ];

  const countryOpts = [{ id: 'United States', country: 'United States' }];

  const { medium, loginCustomer } = useSelector((state) => {
    return { medium: state.customers.medium, loginCustomer: state.customers.loginCustomer };
  });

  const [name, setName] = useState('');
  const [nickName, setNickName] = useState('');
  const [about, setAbout] = useState('');
  const [city, setCity] = useState([]);
  const [state, setState] = useState([]);
  const [country, setCountry] = useState([]);
  const [contactNo, setContactNo] = useState('');

  const [isUploading, setIsUploading] = React.useState(false);

  useEffect(() => {
    if (update) {
      setName(loginCustomer.name);
      setNickName(loginCustomer.nickname);
      setAbout(loginCustomer.about);
      setCity([{ city: loginCustomer.city }]);
      setState([{ state: loginCustomer.state }]);
      setCountry([{ country: loginCustomer.country }]);
      setContactNo(loginCustomer.contact_no);
    }
  }, []);

  const dispatch = useDispatch();
  const history = useHistory();

  const saveCustomerDetails = () => {
    if (
      name == '' ||
      nickName == '' ||
      city == [] ||
      state == [] ||
      country == [] ||
      contactNo == ''
    ) {
      notify({ type: 'error', description: 'Invalid Values' });
      return;
    }
    const data = {
      id: loginDetails.id,
      name,
      nickname: nickName,
      about,
      city: city[0] && city[0].city,
      state: state[0] && state[0].state,
      country: country[0] && country[0].country,
      contact_no: contactNo,
      medium: medium,
    };
    if (update) {
      dispatch(updateCustomer(data, data.id)).then(() => {
        history.push('/restaurants');
      });
    } else {
      dispatch(createCustomer(data)).then(() => {
        history.push('/restaurants');
      });
    }
  };

  const clearMedia = () => {
    dispatch(clearCustomerMedium());
  };

  const mediaUpload = (acceptedFiles, rejectedFiles) => {
    const config = {
      bucketName: window.AWS_BUCKET_NAME,
      region: window.AWS_REGION,
      accessKeyId: window.AWS_ACCESS_ID,
      secretAccessKey: window.AWS_ACCESS_KEY,
      s3Url: 'https://uber-eats-media.s3.amazonaws.com',
    };

    const s3 = new S3(config);

    setIsUploading(true);

    if (acceptedFiles.length > 0) {
      s3.uploadFile(acceptedFiles[0], acceptedFiles[0].name)
        .then((res) => {
          notify({ type: 'info', description: `File ${res.key} uploded...` });

          if (res.status == 204) {
            dispatch(createCustomerMedium({ alt_text: res.key, url: res.location }));
          }
          setIsUploading(false);
        })
        .catch((err) => {
          console.log(err);
          notify({ type: 'error', description: `Uploading File Failed` });
        });
    }
  };

  return (
    <div className={mainContainer}>
      <div className={formContainer}>
        <Display4>Customer Profile</Display4>
        <FormControl label={() => 'Name'}>
          <Input value={name} onChange={(e) => setName(e.target.value)} required />
        </FormControl>
        <FormControl label={() => 'NickName'}>
          <Input value={nickName} onChange={(e) => setNickName(e.target.value)} />
        </FormControl>

        <FormControl label="About You" caption="Tell us something about you">
          <Textarea value={about} onChange={(event) => setAbout(event.currentTarget.value)} />
        </FormControl>

        <FormControl label="Type to search city">
          <Select
            value={city}
            onChange={({ value }) => setCity(value)}
            options={cityOpts}
            labelKey="id"
            valueKey="city"
            required
          />
        </FormControl>

        <FormControl label="Type to search state">
          <Select
            value={state}
            onChange={({ value }) => setState(value)}
            options={stateOpts}
            labelKey="id"
            valueKey="state"
            required
          />
        </FormControl>

        <FormControl label="Type to search country">
          <Select
            value={country}
            onChange={({ value }) => setCountry(value)}
            options={countryOpts}
            labelKey="id"
            valueKey="country"
            required
          />
        </FormControl>

        <FormControl label={() => 'Contact No'}>
          <MaskedInput
            placeholder="Phone number"
            mask="(999) 999-9999"
            value={contactNo}
            onChange={(e) => setContactNo(e.target.value)}
          />
        </FormControl>

        <Button
          type="submit"
          $style={{ alignSelf: 'baseline', width: '30%' }}
          size={SIZE.large}
          onClick={saveCustomerDetails}
        >
          Save
        </Button>
      </div>
      <div className={imgContainer}>
        {medium?.url ? (
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'center',
            }}
          >
            <Card
              overrides={{ Root: { style: { width: '512px' } } }}
              headerImage={medium.url}
              title="Profile Picture"
            ></Card>
            <Button className={css({ margin: '40px' })} onClick={clearMedia}>
              Clear Profile Picture
            </Button>
          </div>
        ) : (
          <FileUploader
            onCancel={() => setIsUploading(false)}
            onDrop={mediaUpload}
            progressMessage={isUploading ? `Uploading... hang tight.` : ''}
          />
        )}
      </div>
    </div>
  );
}

export default CustomerDetails;
