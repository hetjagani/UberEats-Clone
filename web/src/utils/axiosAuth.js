import axios from 'axios';
import { getCookie, setCookie } from 'react-use-cookie';
import notify from './notify';

function createAxiosAuthMiddleware() {
  return ({ getState }) =>
    (next) =>
    (action) => {
      const authToken = getCookie('auth');

      if (authToken) {
        axios.defaults.headers.common['Authorization'] = authToken;
      }

      axios.defaults.baseURL = window.BACKEND_API_URL;
      axios.defaults.withCredentials = false;
      axios.interceptors.response.use(
        (res) => res,
        (err) => {
          if (err.response.status === 401) {
            setCookie('auth', null);
            window.location = '/auth/login';
            notify({ type: 'error', description: 'Unauthorized. Please Login.' });
          }
          return Promise.reject(err);
        },
      );
      return next(action);
    };
}

export default createAxiosAuthMiddleware;
