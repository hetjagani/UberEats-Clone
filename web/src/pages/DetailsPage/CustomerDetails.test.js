import { shallow } from 'enzyme';
import React from 'react';
import '../../utils/testsetup';
import CustomerDetails from './CustomerDetails';
import { Display4 } from 'baseui/typography';

import * as redux from 'react-redux';

describe('Customer Details Page Test', () => {
  const initialState = {
    medium: [],
    loginCustomer: {
      name: 'Test User',
      nickname: 'TU',
      about: 'Test About',
      city: 'Fremont',
      state: 'California',
      country: 'USA',
      contact_no: '9898989898',
    },
  };

  const spy = jest.spyOn(redux, 'useSelector');
  spy.mockReturnValue(initialState);
  const spy2 = jest.spyOn(redux, 'useDispatch');
  spy2.mockReturnValue(null);

  it('should render Customer Details page', () => {
    const detailsPage = shallow(<CustomerDetails />);
    expect(detailsPage.getElements()).toMatchSnapshot();
  });

  it('should have Customer Profile heading', () => {
    const detailsPage = shallow(<CustomerDetails />);
    expect(detailsPage.find(Display4).text()).toBe('Customer Profile');
  });
});
