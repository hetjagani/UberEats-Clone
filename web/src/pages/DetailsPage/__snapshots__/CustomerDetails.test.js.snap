// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Details Page Test should render Customer Details page 1`] = `
Array [
  <div
    className=""
  >
    <div
      className=""
    >
      <DisplayXSmall>
        Customer Profile
      </DisplayXSmall>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <Input
          autoComplete="on"
          autoFocus={false}
          clearable={false}
          disabled={false}
          endEnhancer={null}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          overrides={Object {}}
          required={true}
          size="default"
          startEnhancer={null}
          type="text"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <Input
          autoComplete="on"
          autoFocus={false}
          clearable={false}
          disabled={false}
          endEnhancer={null}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          overrides={Object {}}
          required={false}
          size="default"
          startEnhancer={null}
          type="text"
          value=""
        />
      </FormControl>
      <FormControl
        caption="Tell us something about you"
        disabled={false}
        label="About You"
        overrides={Object {}}
      >
        <Textarea
          autoFocus={false}
          disabled={false}
          error={false}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          overrides={Object {}}
          placeholder=""
          required={false}
          rows={3}
          size="default"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search city"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "city": "San Jose",
                "id": "San Jose",
              },
              Object {
                "city": "Berkely",
                "id": "Berkely",
              },
              Object {
                "city": "Los Angeles",
                "id": "Los Angeles",
              },
              Object {
                "city": "San Diego",
                "id": "San Diego",
              },
              Object {
                "city": "San Francisco",
                "id": "San Francisco",
              },
              Object {
                "city": "Fresno",
                "id": "Fresno",
              },
              Object {
                "city": "Sacremento",
                "id": "Sacremento",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="city"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search state"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": "California",
                "state": "California",
              },
              Object {
                "id": "Arizona",
                "state": "Arizona",
              },
              Object {
                "id": "Colorado",
                "state": "Colorado",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="state"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search country"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "country": "United States",
                "id": "United States",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="country"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <MaskedInput
          mask="(999) 999-9999"
          maskChar=" "
          onChange={[Function]}
          placeholder="Phone number"
          value=""
        />
      </FormControl>
      <Button
        $style={
          Object {
            "alignSelf": "baseline",
            "width": "30%",
          }
        }
        onClick={[Function]}
        size="large"
        type="submit"
      >
        Save
      </Button>
    </div>
    <div
      className=""
    >
      <FileUploader
        disableClick={true}
        onCancel={[Function]}
        onDrop={[Function]}
        overrides={Object {}}
        progressMessage=""
      />
    </div>
  </div>,
]
`;
